src/services/AudioMixer.cxx

```diff
@@ -218,9 +364,37 @@ void AudioMixer::loadSoundtrack(Iterator begin, Iterator end, CallbackType creat
           throw RuntimeError(ErrorMsg);
         }
         m_Soundtracks.emplace_back(createSoundtrack(name, chunk));
-        for(AudioTrigger trigger : soundtrack.triggers)
-          m_Triggers[trigger].emplace_back(name);
-    }
+        
+        #ifdef USE_OPENAL_SOFT
+      
+      //initialize buffer
+	  alGenBuffers(1, &m_Soundtracks.back()->buffer);
+	  assert(alGetError() == AL_NO_ERROR && "Could not create buffers");
+      //set buffer data
+	  //alBufferData(buffer, format, data, slen, frequency);
+	  ALenum format;
+	  if(Settings::instance().audio3DStatus){format = AL_FORMAT_MONO16;}
+	  else{format = AL_FORMAT_STEREO16;}
+	  alBufferData(m_Soundtracks.back()->buffer, format, m_Soundtracks.back()->Chunks->abuf, m_Soundtracks.back()->Chunks->alen, 44100);
+      
+      //initialize source
+      alGenSources(1, &m_Soundtracks.back()->source);
+	  alSourcei(m_Soundtracks.back()->source, AL_SOURCE_RELATIVE, AL_FALSE);
+	  assert(alGetError()==AL_NO_ERROR && "Failed to setup sound source.");
```

Comment:
Replace by exception
```suggestion
	  errorCode = alGetError();
	  if(errorCode != AL_NO_ERROR)
	    throw RuntimeError("Failed to setup sound source: Error " + std::to_string(errorCode));
```

Reply:
Resolved. throwing am exception instead of asserting.